core/src/main/java/org/apache/iceberg/IncrementalFileCleanup.java

```diff
@@ -260,10 +260,18 @@ public void cleanFiles(TableMetadata beforeExpiration, TableMetadata afterExpira
         findFilesToDelete(manifestsToScan, manifestsToRevert, validIds, afterExpiration);
 
     deleteFiles(filesToDelete, "data");
-    LOG.warn("Manifests to delete: {}", Joiner.on(", ").join(manifestsToDelete));
-    LOG.warn("Manifests Lists to delete: {}", Joiner.on(", ").join(manifestListsToDelete));
+    LOG.debug("Manifests to delete: {}", Joiner.on(", ").join(manifestsToDelete));
```

Comment:
Joining should not be performed prior to passing the argument to debug, or else it will always create a string regardless of whether it is used.